src/Simplify_Internal.h

```diff
@@ -258,7 +258,7 @@ class Simplify : public VariadicVisitor<Simplify, Expr, Stmt> {
 
         std::vector<const Variable *> pop_list;
         std::vector<const Variable *> bounds_pop_list;
-        std::vector<Expr> truths, falsehoods;
+        std::set<Expr, IRDeepCompare> truths, falsehoods;
```

Comment:
comparing Exprs for < is actually a lot cheaper than hashing them, because you can early out as soon as you find an IRNodeType that differs instead of having to descend to the bottom.

Reply:
Might be worth putting that in a comment then :-)

Reply:
Meh, that was a bit of a guess. Nowhere in the compiler do we use unordered sets or maps with Expr keys, because we don't have a hash function. I'm just doing what we do everywhere else.